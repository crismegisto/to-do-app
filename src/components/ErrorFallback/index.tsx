import React from 'react';
import { Text, View, Button } from 'react-native';
import { FallbackProps } from 'react-error-boundary';

import styles from './styles';

function ErrorFallback({ error, resetErrorBoundary }: FallbackProps) {
  return (
    <View style={styles.container}>
      <Text>{error.message}</Text>
      <Button title="Try Again" onPress={resetErrorBoundary} />
    </View>
  );
}

export default ErrorFallback;
