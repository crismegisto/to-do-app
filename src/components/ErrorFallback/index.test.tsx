import React from 'react';
import { screen } from '@testing-library/react-native';

import { renderWithProvider, initialStoreState } from '../../../utils/test/redux';
import ErrorFallback from './index';

const error = {
  name: 'error',
  message: 'Mock error'
};

describe('<ErrorFallback /> Tests', () => {
  it('match snapshot and button name', () => {
    renderWithProvider(<ErrorFallback error={error} resetErrorBoundary={() => jest.fn()} />, {
      preloadedState: initialStoreState
    });

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
