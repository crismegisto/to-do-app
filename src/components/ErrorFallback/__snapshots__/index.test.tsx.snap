// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorFallback /> Tests match snapshot and button name 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Mock error
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Try Again
      </Text>
    </View>
  </View>
</View>
`;
